### PR TITLE
Feature Sign-up. Validation & Handling

### PR DESCRIPTION
**Pull request by hoangminhtung**
image
image
image

### CODE REVIEW

**Comment by hoangminhtung** [comment]
ready

File: src/auth/dto/auth-credentials.dto.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export class AuthCredentialsDto {
+  @IsString()
+  @MinLength(4)
+  @MaxLength(20)
+  readonly username: string;
+
+  @IsString()
+  @MinLength(8)
+  @MaxLength(32)
+  @Matches(/((?=.*\d)|(?=.*\W+))(?![.\n])(?=.*[A-Z])(?=.*[a-z]).*$/, {
```
Comment (huongnt-2545) [inline src/auth/dto/auth-credentials.dto.ts:12]: Tạo file `/src/utils/constants.ts` để khai báo các constant cần dùng nhé

Comment (hoangminhtung) [reply src/auth/dto/auth-credentials.dto.ts]: dạ chị

**Comment by hoangminhtung** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**